tests/handle_error/run.sh

```diff
@@ -0,0 +1,394 @@
+    run_sql_source2 "alter table ${db}.${tb2} CHARACTER SET latin1 COLLATE latin1_danish_ci;"
+    # 21 second ddl
+    run_sql_source2 "alter table ${db}.${tb1} add column new_col1 varchar(10);"
+    run_sql_source2 "insert into ${db}.${tb1} values(6,'bbb');"
+    # 12/22 second ddl
+    run_sql_source1 "alter table ${db}.${tb2} add column new_col1 varchar(10);"
+    run_sql_source1 "insert into ${db}.${tb2} values(7,'ccc');"
+    run_sql_source2 "alter table ${db}.${tb2} add column new_col1 varchar(10);"
+    run_sql_source2 "insert into ${db}.${tb2} values(8,'ddd');"
+
+    # begin to handle error
+    # 11/21 first ddl: unsupport error
+    run_dm_ctl_with_retry $WORK_DIR "127.0.0.1:$MASTER_PORT" \
+            "query-status test" \
+            "Unsupported modify change collate from latin1_bin to latin1_danish_ci" 2
+
+    # skip 11/21 first ddl
+    run_dm_ctl $WORK_DIR "127.0.0.1:$MASTER_PORT" \
+            "handle-error test skip" \
+            "\"result\": true" 3
+
+    if [[ "$1" = "pessimistic" ]]; then
+        # 11 second ddl bypass, 12 first ddl: detect conflict
+        # 22 first ddl: unsupport error
+        run_dm_ctl_with_retry $WORK_DIR "127.0.0.1:$MASTER_PORT" \
+                "query-status test" \
+                "detect inconsistent DDL sequence from source" 1 \
+                "Unsupported modify change collate from latin1_bin to latin1_danish_ci" 1
+    else
+        # 12/22 first ddl: unsupport error
+        run_dm_ctl_with_retry $WORK_DIR "127.0.0.1:$MASTER_PORT" \
+                "query-status test" \
+                "Unsupported modify change collate from latin1_bin to latin1_danish_ci" 2
+    fi
+
+    run_dm_ctl $WORK_DIR "127.0.0.1:$MASTER_PORT" \
+            "handle-error test skip -s mysql-replica-01,mysql-replica-02" \
+            "\"result\": true" 3
+
+    run_dm_ctl_with_retry $WORK_DIR "127.0.0.1:$MASTER_PORT" \
+            "query-status test" \
+            "\"stage\": \"Running\"" 2
+
+    run_sql_tidb_with_retry "select count(1) from ${db}.${tb}" "count(1): 8"
+}
+
+function DM_SKIP_ERROR_SHARDING() {
+    run_case SKIP_ERROR_SHARDING "double-source-pessimistic" "init_table 111 112 211 212" "clean_table" "pessimistic"
+    run_case SKIP_ERROR_SHARDING "double-source-optimistic" "init_table 111 112 211 212" "clean_table" "optimistic"
+}
+
+# replace add foreign key with database name
+# two source, no sharding
+function DM_REPLACE_ERROR_CASE() {
+    run_sql_source1 "insert into ${db}.${tb2} values(1,1);"
+    run_sql_source1 "insert into ${db}.${tb1} values(2,1);"
+
+    # error in TiDB
+    run_sql_source1 "alter table ${db}.${tb1} add constraint fk foreign key (b) references ${tb2}(a);"
+    run_sql_source1 "insert into ${db}.${tb2} values(3,3);"
+    run_sql_source1 "insert into ${db}.${tb1} values(4,3);"
+
+    run_dm_ctl_with_retry $WORK_DIR "127.0.0.1:$MASTER_PORT" \
+            "query-status test" \
+            "No database selected" 1
```

Comment:
this case may need to update someday after updated TiDB dependency. 😂

Reply:
Any typical case? 😅